Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsFragmentPresenter.java

```diff
@@ -304,119 +269,74 @@ private void addAudioSettings(ArrayList<SettingsItem> sl)
+    sl.add(new SingleChoiceSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_GAME_CUBE_LANGUAGE, R.string.gamecube_system_language, 0,
+            R.array.gameCubeSystemLanguageEntries, R.array.gameCubeSystemLanguageValues, 0));
+    sl.add(new SingleChoiceSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_SLOT_A_DEVICE, R.string.slot_a_device, 0, R.array.slotDeviceEntries,
+            R.array.slotDeviceValues, 8));
+    sl.add(new SingleChoiceSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_SLOT_B_DEVICE, R.string.slot_b_device, 0, R.array.slotDeviceEntries,
+            R.array.slotDeviceValues, 255));
   }
 
   private void addWiiSettings(ArrayList<SettingsItem> sl)
   {
-    Setting wiiSDCard = null;
-    Setting wiiSDWrites = null;
-    Setting continuousScan = null;
-    Setting wiimoteSpeaker = null;
-
-    SettingSection coreSection = mSettings.getSection(Settings.SECTION_INI_CORE);
-    wiiSDCard = coreSection.getSetting(SettingsFile.KEY_WII_SD_CARD);
-    wiiSDWrites = coreSection.getSetting(SettingsFile.KEY_WII_SD_CARD_ALLOW_WRITES);
-    continuousScan = coreSection.getSetting(SettingsFile.KEY_WIIMOTE_SCAN);
-    wiimoteSpeaker = coreSection.getSetting(SettingsFile.KEY_WIIMOTE_SPEAKER);
-
-    sl.add(new CheckBoxSetting(SettingsFile.KEY_WII_SD_CARD, Settings.SECTION_INI_CORE,
-            R.string.insert_sd_card, R.string.insert_sd_card_description, true, wiiSDCard));
-    sl.add(new CheckBoxSetting(SettingsFile.KEY_WII_SD_CARD_ALLOW_WRITES, Settings.SECTION_INI_CORE,
-            R.string.wii_sd_card_allow_writes, 0, true, wiiSDWrites));
-    sl.add(new CheckBoxSetting(SettingsFile.KEY_WIIMOTE_SCAN, Settings.SECTION_INI_CORE,
-            R.string.wiimote_scanning, R.string.wiimote_scanning_description, true,
-            continuousScan));
-    sl.add(new CheckBoxSetting(SettingsFile.KEY_WIIMOTE_SPEAKER, Settings.SECTION_INI_CORE,
-            R.string.wiimote_speaker, R.string.wiimote_speaker_description, true, wiimoteSpeaker));
+    sl.add(new CheckBoxSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_WII_SD_CARD, R.string.insert_sd_card,
+            R.string.insert_sd_card_description, true));
+    sl.add(new CheckBoxSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_WII_SD_CARD_ALLOW_WRITES, R.string.wii_sd_card_allow_writes, 0, true));
+    sl.add(new CheckBoxSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_WIIMOTE_SCAN, R.string.wiimote_scanning,
+            R.string.wiimote_scanning_description, true));
+    sl.add(new CheckBoxSetting(SettingsFile.FILE_NAME_DOLPHIN, Settings.SECTION_INI_CORE,
+            SettingsFile.KEY_WIIMOTE_SPEAKER, R.string.wiimote_speaker,
+            R.string.wiimote_speaker_description, true));
```

Comment:
When Dolphin.ini is created, `WiimoteContinuousScanning` and `WiimoteEnableSpeaker` default to false. Should the defaults be false here too?

Reply:
This is fixed in PR #8975, but in case you want the fix early, I've submitted it as PR #9024.